roles/zabbix_agent/tasks/main.yml

```diff
@@ -118,6 +118,7 @@
 - name: "Run the API calls to Zabbix Server"
   vars:
     gather_facts: false
+    ansible_host: "{{ zabbix_api_server_host }}"
```

Comment:
So I have two points here.

1.  Please reference the conversation in https://github.com/ansible-collections/community.zabbix/pull/920 about using delegate_to vs ansible_host.
2. @D3DeFi  and @BGmot I have a question.  I'll admit I still don't totally understand how the HTTPAPI works.  My question is, why do we have to set vars (i.e. `ansible_httpapi_use_ssl` in ./tasks/main line 118 but we don't have to do the same thing over in the api.yml calls?

Reply:
I have tried several solutions and this was the only thing that fixed the problem of the API calls going to the monitored server instead of `zabbix_api_server_host`

Reply:
@pyrodie18 you need these variables set for any module from this collection (API call). Ansible is flexible enough to set variables at different places (I personally set them up in my inventory for Zabbix server host). Here we set them up at `ansible.builtin.include_tasks` task because they will be used for every task in api.yml (just to avoid duplication).
We don't need this line 121 as every task in api.yml has `delegate_to`
https://github.com/ansible-collections/community.zabbix/blob/main/roles/zabbix_agent/tasks/api.yml#L41
https://github.com/ansible-collections/community.zabbix/blob/main/roles/zabbix_agent/tasks/api.yml#L57

Reply:
I see the `delegate_to` but still in my setup it is not working without the `ansible_host` line.

Reply:
Please troubleshoot your set up. We can help -) But it is definitely working.

Reply:
> @pyrodie18 you need these variables set for any module from this collection (API call). Ansible is flexible enough to set variables at different places (I personally set them up in my inventory for Zabbix server host). Here we set them up at `ansible.builtin.include_tasks` task because they will be used for every task in api.yml (just to avoid duplication). We don't need this line 121 as every task in api.yml has `delegate_to` https://github.com/ansible-collections/community.zabbix/blob/main/roles/zabbix_agent/tasks/api.yml#L41 https://github.com/ansible-collections/community.zabbix/blob/main/roles/zabbix_agent/tasks/api.yml#L57

Oh duh...that's what I get for looking at code at midnight.

Reply:
@BGmot Even when I hard-code the hostname of my Zabbix server 3x in the `delegate_to` in `zabbix_agent/tasks/api.yml` the HttpApi still connects to the monitored node instead of the Zabbix server. When I use `ansible_host` like in this PR, it works.

Reply:
Ok, let me test more. I'll get back to you on this.

Reply:
I noticed when delegating in the `api.xml` the Ansible output contains:
```
fatal: [someserver.mydomain.io -> zabbix.mydomain.io(None)]: ....
```
while another delegation when it works does not contain the `(None)`:
```
ok: [someserver.mydomain.io -> delegated-to.mydomain.io]
```

Reply:
On a template of an issue creation we have environment details to be provided. It's not for fun. It's for us to waste less time to re-produce your problem and try to resolve it. Please provide Python version, Ansible version, and OS you are running Ansible controller on.

Reply:
Found https://github.com/ansible/ansible/issues/27351
Indeed it works when I don't set `ansible_ssh_host` for my Zabbix host.
I still think you should consider switching to overriding `ansible_host` instead using `delegate_to` because you want to use the HTTP API to directly connect to the `zabbix_api_server_host` and you don't need any Ansible connection magic to be involved.

```
Python 3.12.3
ansible [core 2.16.7]
  python version = 3.12.3 (main, Apr  9 2024, 08:09:14) [Clang 15.0.0 (clang-1500.3.9.4)] (/opt/homebrew/Cellar/ansible/9.6.0/libexec/bin/python)
  jinja version = 3.1.4
Darwin MyHost.local 23.5.0 Darwin Kernel Version 23.5.0: Wed May  1 20:14:38 PDT 2024; root:xnu-10063.121.3~5/RELEASE_ARM64_T6020 arm64
macOS Sonoma 14.5
```

Reply:
The ansible issue you found claimed to be fixed long time ago.
I don't know I am comfortable with the code we have now and nobody complained about this so far. Looks like the issue is specific to your use case. I am not very comfortable with proposed changed. @pyrodie18, @D3DeFi what do you think?

Reply:
> The ansible issue you found claimed to be fixed long time ago. I don't know I am comfortable with the code we have now and nobody complained about this so far. Looks like the issue is specific to your use case. I am not very comfortable with proposed changed. @pyrodie18, @D3DeFi what do you think?

Agree.  We made this change a long time ago and no one else has complained about it.  Given that while I appreciate the effort with the PR, I'm inclined to not pull it in.

Reply:
OK I will remove this change. 
I can work around it by **not** setting `ansible_ssh_host` in the host_vars of my Zabbix server.
If you would like to repeat the problem: Just add this setting with the IP address of the Zabbix server.